src/form/__tests__/analytics.test.tsx

```diff
@@ -0,0 +1,144 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import createWrapper from '../../../lib/components/test-utils/dom';
+import Button from '../../../lib/components/button';
+import Form from '../../../lib/components/form';
+
+import { FunnelMetrics, setFunnelMetrics } from '../../../lib/components/internal/analytics';
+
+const mockedFunnelInteractionId = 'mocked-funnel-id';
+function mockFunnelMetrics() {
+  setFunnelMetrics({
+    funnelStart: jest.fn(() => mockedFunnelInteractionId),
+    funnelError: jest.fn(),
+    funnelComplete: jest.fn(),
+    funnelSuccessful: jest.fn(),
+    funnelCancelled: jest.fn(),
+    funnelStepStart: jest.fn(),
+    funnelStepComplete: jest.fn(),
+    funnelStepNavigation: jest.fn(),
+    funnelSubStepStart: jest.fn(),
+    funnelSubStepComplete: jest.fn(),
+    funnelSubStepError: jest.fn(),
+    helpPanelInteracted: jest.fn(),
+    externalLinkInteracted: jest.fn(),
+  });
```

Comment:
You're using this mock in at least 3 places already. Might be worth moving it to a shared file.

Reply:
Moved to a common folder in the analytics folder.